api/utils/make_appointment.py

```diff
@@ -306,7 +309,10 @@ def _handle_default_criteria(
         else:
             messages.append(DEFAULT_INVALID_MESSAGES[failed_criterion])
 
-    if should_exclude_by_age(participant, default_criteria):
+    if time_slot and should_exclude_by_age(participant, default_criteria, time_slot.datetime.date()):
+        failed_criteria.append('age')
+        messages.append(DEFAULT_INVALID_MESSAGES['age'])
+    elif not time_slot and should_exclude_by_age(participant, default_criteria):
         failed_criteria.append('age')
         messages.append(DEFAULT_INVALID_MESSAGES['age'])
```

Comment:
This has a lot of (unecessary) duplication, and I feel like the actual reason for the two ifs is a bit hidden this way.

I would prefer something like this:
```suggestion
    test_date = time_slot.datetime.date() if time_slot else date.today()
    if should_exclude_by_age(participant, default_criteria, test_date):
        failed_criteria.append('age')
        messages.append(DEFAULT_INVALID_MESSAGES['age'])
```

I'd also be fine with splitting that `x if y else z` into seperate lines if that's what you prefer